File: android/src/main/java/com/getcapacitor/community/classes/results/StatusResult.java
package com.getcapacitor.community.classes.results;

import com.getcapacitor.JSObject;
import com.getcapacitor.community.interfaces.Result;

public class StatusResult implements Result {

    private final boolean isAdvertising;
    private final boolean isDiscovering;

    public StatusResult(boolean isAdvertising, boolean isDiscovering) {
        this.isAdvertising = isAdvertising;
        this.isDiscovering = isDiscovering;
    }

    @Override
    public JSObject toJSObject() {
        JSObject result = new JSObject();

        result.put("isAdvertising", isAdvertising);
        result.put("isDiscovering", isDiscovering);

        return result;
    }
}

File: ios/Sources/NearbyPlugin/Classes/Results/StatusResult.swift
import Foundation
import Capacitor

@objc public class StatusResult: NSObject, Result {
    let isAdvertising: Bool
    let isDiscovering: Bool

    init(isAdvertising: Bool, isDiscovering: Bool) {
        self.isAdvertising = isAdvertising
        self.isDiscovering = isDiscovering
    }

    public func toJSObject() -> AnyObject {
        var result = JSObject()

        result["isAdvertising"] = isAdvertising
        result["isDiscovering"] = isDiscovering

        return result as AnyObject
    }
}
